tests/e2e/transaction/transactionBuilder.test.ts

```diff
@@ -554,6 +555,102 @@ describe("transaction builder", () => {
       expect(signedTransaction instanceof SignedTransaction).toBeTruthy();
     });
   });
+
+  describe("generateTransactionHash", () => {
+    test("it generates a single signer signed transaction hash", async () => {
+      const alice = Account.generate();
+      await aptos.fundAccount({ accountAddress: alice.accountAddress, amount: FUND_AMOUNT });
+      const transaction = await aptos.coin.transferCoinTransaction({
+        sender: alice.accountAddress,
+        recipient: "0x1",
+        amount: 1,
+      });
+
+      const senderAuthenticator = sign({ signer: alice, transaction });
+
+      // Generate hash
+      const signedTxnInput = {
+        transaction,
+        senderAuthenticator,
+      };
+      const transactionHash = generateTransactionHash(signedTxnInput);
+
+      // Submit transaction
+      const submitted = await aptos.transaction.submit.simple(signedTxnInput);
```

Comment:
do we need to actually submit it to get the transaction hash or is it enough to simulate?

Reply:
Simulation would give a different hash, because the signature is zeroed out.